Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift

```diff
@@ -273,40 +273,37 @@ public class ByteBuffer: Codable {
     }
 }
 
-extension ByteBuffer: AwsStream {
-    public var status: aws_stream_status {
-        return aws_stream_status(is_end_of_stream: self.currentIndex == array.count, is_valid: true)
-    }
+extension ByteBuffer: IStreamable {
 
-    public var length: UInt {
-        return UInt(array.count)
+    public func length() -> UInt64 {
+        return UInt64(array.count)
     }
 
-    public func seek(offset: Int64, basis: aws_stream_seek_basis) -> Bool {
-        let targetOffset: Int64
-        if basis.rawValue == AWS_SSB_BEGIN.rawValue {
-            targetOffset = offset
+    public func seek(offset: Int64, streamSeekType: StreamSeekType) throws {
+        if abs(offset) > array.count
+                   || (offset < 0 && streamSeekType == .begin)
+                   || (offset > 0 && streamSeekType == .end) {
+            throw CommonRunTimeError.crtError(CRTError(code: AWS_IO_STREAM_INVALID_SEEK_POSITION.rawValue))
+        }
 
-        } else {
-            targetOffset = Int64(length) - offset
+        switch streamSeekType {
+        case .begin:
+            currentIndex = Int(offset)
+        case .end:
+            currentIndex = array.count + Int(offset)
         }
-        currentIndex = Int(targetOffset)
-        return true
     }
 
-    public func read(buffer: inout aws_byte_buf) -> Bool {
-        let bufferCapacity = buffer.capacity - buffer.len
-        let arrayEnd = (bufferCapacity + self.currentIndex) < array.count ? bufferCapacity + self.currentIndex : array.count
-        let dataArray = array[self.currentIndex..<(arrayEnd)]
-        if dataArray.count > 0 {
-            let result = buffer.buffer.advanced(by: buffer.len)
-            let resultBufferPointer = UnsafeMutableBufferPointer.init(start: result, count: dataArray.count)
-            dataArray.copyBytes(to: resultBufferPointer)
-            self.currentIndex = arrayEnd
-            buffer.len += dataArray.count
-            return true
+    public func read(buffer: UnsafeMutableBufferPointer<UInt8>) -> Int? {
+        let endIndex = self.currentIndex + min(array.count - self.currentIndex, buffer.count)
```

Comment:
Nit: It'd be great to stay consistent with our usage of `self`. Can we either change `self.currentIndex` in this function to just `currentIndex` or update the contents of `func seek(offset: Int64, streamSeekType: StreamSeekType)` to use `self.currentIndex`?

Reply:
Thanks, fixed.